feat(loadtest): Support shared bridge interface in loadtest EVM-571

ly0va opened the pull request:
## What ❔

Adds support for the contracts' new post-shared bridge interface in loadtest.

## Why ❔

To be able to run loadtest against custom hyperchains (including with custom base token), and not only Era.


## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

ly0va committed:
Works with new interface

ly0va committed:
Keep compatibility

ly0va committed:
Merge remote-tracking branch 'origin/main' into lyova-fix-loadtest

ly0va committed:
Move CI to new interface

ly0va committed:
Support custom base token chains

ly0va committed:
Merge remote-tracking branch 'origin/main' into lyova-fix-loadtest

ly0va committed:
Post-merge fixes

ly0va committed:
Drop legacy support

ly0va committed:
Merge branch 'main' into lyova-fix-loadtest

StanislavBreadless commented on core/tests/loadnext/src/executor.rs:
```diff
@@ -85,11 +108,32 @@ impl Executor {
 
-        self.deposit_eth_to_paymaster().await?;
+        tracing::info!("Master Account: Minting Test ERC20 token...");
+        self.mint(self.config.main_token).await?;
```
nit: it might be sometimes confusing to the reader on what is the difference between "main" and "base" tokens. I would suggest renaming the "main" token field

ly0va replied:
It's used in workflows (e.g. [here](https://github.com/matter-labs/zksync-era-workflows/blob/5d959660d8a9c439b45a4f4d9c27af7993c6e035/.github/workflows/loadtest-reusable.yml#L123)) so probably easier to add a doc comment?

StanislavBreadless commented on core/tests/loadnext/src/executor.rs:
```diff
@@ -280,23 +324,36 @@ impl Executor {
             .get_balance(paymaster_address, None)
             .await?;
 
+        // TODO: decimals may differ from eth
```
Better to create a task (if you believe it is to be resolved) or just state that in comment that the formatting supports only 18 decimals for now

StanislavBreadless commented on core/tests/loadnext/src/sdk/ethereum/mod.rs:
```diff
@@ -524,29 +566,43 @@ impl<S: EthereumSigner> EthereumProvider<S> {
                 U256::from(L1_TO_L2_GAS_PER_PUBDATA),
+                Address::default(),
+                bridge_address,
+                U256::zero(),
```
maybe worth adding a comment that we do not support depositing ETH to non-eth-based chains

Deniallugo committed:
Update contracts

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Update wallets

Signed-off-by: Danil <deniallugo@gmail.com>

ly0va committed:
Add a bunch of comments

Deniallugo committed:
Initialize with zk stack

Signed-off-by: Danil <deniallugo@gmail.com>

ly0va committed:
Merge branch 'main' into lyova-fix-loadtest

ly0va committed:
Fix conlifts

Deniallugo committed:
Merge branch 'main' into deniallugo-init-with-new-stack

kelemeno commented on core/tests/loadnext/src/sdk/ethereum/mod.rs:
```diff
@@ -185,7 +213,6 @@ impl<S: EthereumSigner> EthereumProvider<S> {
         bridge: Option<Address>,
     ) -> Result<Address, ClientError> {
-        // TODO(EVM-571): This should be moved to the shared bridge, which does not have `l2_token_address` on L1. Use L2 contracts instead.
         let bridge = bridge.unwrap_or(self.default_bridges.l1_erc20_default_bridge.unwrap());
```
Would it be possible to go via the L2SharedBridge? That would be a better long term solution. The L1 erc20 bridge is deprecated, we should avoid using it

ly0va replied:
we don't use this method anymore, I can remove it

kelemeno commented on core/tests/loadnext/src/sdk/ethereum/mod.rs:
```diff
@@ -439,6 +479,7 @@ impl<S: EthereumSigner> EthereumProvider<S> {
 
     /// Performs a deposit in zkSync network.
     /// For ERC20 tokens, a deposit must be approved beforehand via the `EthereumProvider::approve_erc20_token_deposits` method.
+    /// Cannot deposit ETH onto custom base token chains.
```
We can deposit Eth to custom base token chains though. I am not sure what you meant here?

ly0va replied:
I meant this method can't.

aon committed:
feat: add zk-toolbox

aon committed:
feat: remove unnecessary files from zk_toolbox

aon committed:
Merge branch 'main' into aon-add-zk-toolbox

ly0va committed:
Clean up

aon committed:
fix: add missing rust-toolchain for zk_toolbox

slowli committed:
fix(en): Minor node fixes (#1978)

## What ❔

Fixes minor issues with the EN:

- Transient error detection for RPC clients is suboptimal, leading to
what really is a transient error crashing consistency checker.
- Since refactoring signal handling, `ManagedTasks::wait_single()` on EN
may inappropriately log task termination errors even after a signal is
being received.
- Since recent refactoring (?), block fetcher doesn't tag its L2 client.
- Block fetcher may produce many rate limiting logs, which currently
have WARN level.
- Initializing L2 client multiple times (e.g., in a load test) produces
many WARN logs.

## Why ❔

Improves EN UX.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [x] Spellcheck has been run via `zk spellcheck`.

aon committed:
Merge remote-tracking branch 'origin/deniallugo-init-with-new-stack' into aon-add-zk-toolbox-ci

ly0va committed:
More cleanup

ly0va committed:
Merge remote-tracking branch 'origin/main' into lyova-fix-loadtest

aon committed:
feat: add zk toolbox ci

Deniallugo committed:
Return prereq check

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Remove docker-comopose from prereq

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Move building toolbox

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Fix

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Add prints

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
fifx test

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Print pre save

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
echo pwd

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Use non relative paths

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
remove file prepend

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Prepend without tempfiles

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Fix ci

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Fix l1 rpc url

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Remove slow

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
use inception for running server

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
On a way to run loadtest

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Merge branch 'aon-add-zk-toolbox-ci' into lyova-fix-loadtest

Deniallugo committed:
Change default chain id

Signed-off-by: Danil <deniallugo@gmail.com>

deniallugo-ml commented:
Closing this PR as it targets a protocol version that has since been superseded. ZKsync Era is currently on protocol v31, and the upgrade calldata / configuration / constants in this PR were specific to an earlier version that has already shipped or been replaced.

Thank you for the work here — the approach may still be useful as a reference for future upgrades.
